src/objects/Mesh.js

```diff
@@ -49,6 +49,55 @@ class Mesh extends Object3D {
 
 	}
 
+	volume( precision ) {
+
+		let volume = 0;
+		const vertices = this.geometry.attributes.position.array;
+		let indexes = this.geometry.index ? this.geometry.index.array : null;
+
+		function triangularVolume( p1, p2, p3 ) {
+
+			const v321 = p3.x * p2.y * p1.z;
+			const v231 = p2.x * p3.y * p1.z;
+			const v312 = p3.x * p1.y * p2.z;
+			const v132 = p2.x * p1.y * p3.z;
+			const v213 = p1.x * p3.y * p2.z;
+			const v123 = p1.x * p2.y * p3.z;
+			return ( 1.0 / 6.0 ) * ( - v321 + v231 + v312 - v132 - v213 + v123 );
+
+		}
+
+		if ( ! indexes ) {
+
+			indexes = Array.from( { length: vertices.length / 3 }, ( _, i ) => i );
+
+		}
+
+		const vectorA = new Vector3();
+		const vectorB = new Vector3();
+		const vectorC = new Vector3();
+
+		for ( let i = 0; i < indexes.length; i += 3 ) {
+
+			vectorA.fromBufferAttribute( vertices, indexes[ i ] * 3 );
```

Comment:
Please do not directly access the index _array_. I suggest you write a separate loop if no index is present:
```js
if ( index ) {

    for ( let i = 0; i < index.count; i += 3 ) {

        vectorA.fromBufferAttribute( vertices, index.getX( i ) );
        vectorB.fromBufferAttribute( vertices, index.getX( i + 1 ) );
        vectorC.fromBufferAttribute( vertices, index.getX( i + 2 ) );
	volume += triangularVolume( vectorA, vectorB, vectorC );

    }

} else {

    for ( let i = 0; i < positionAttribute.count; i += 3 ) {

        vectorA.fromBufferAttribute( vertices, i );
        vectorB.fromBufferAttribute( vertices, i + 1 );
        vectorC.fromBufferAttribute( vertices, i + 2 );
        volume += triangularVolume( vectorA, vectorB, vectorC );

    }

}
```